eclass/elisp-common.eclass

```diff
@@ -287,6 +298,34 @@ elisp-need-emacs() {
 	return 0
 }
 
+# @FUNCTION: elisp-emacs-nativecomp
+# @RETURN: exit status of Emacs
+# @DESCRIPTION:
+# Output whether currently active Emacs was built with nativecomp.
+
+elisp-emacs-nativecomp() {
+	local cmd
+	if ! ver_test "${_ELISP_EMACS_VERSION}" -ge "${_EMACS_MAYBE_NATIVE_COMP}"; then
+		return 1
+	fi
+	cmd='(or (string-match-p "--with-nativecomp" system-configuration-options) (error "nativecomp not enabled"))'
+	return ${EMACS} ${EMACSFLAGS} -l "${cmd}"
```

Comment:
That test looks complicated. How about this instead:
```
    [[ $(${EMACS} ${EMACSFLAGS} --eval "(princ (fboundp 'batch-native-compile))") == t ]]
```

Reply:
I don't think that is entirely safe given the checks that @AndreaCorallo added in https://github.com/emacs-mirror/emacs/commit/6c108e44c9522d1a70ac49c4810ed5927b8b2223. But having now looked at that commit more closely it looks like calling `comp-ensure-native-compiler` would be the right thing to do here.